src/main/java/org/example/models/JobRoleRequest.java

```diff
@@ -0,0 +1,115 @@
+package org.example.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.sql.Date;
+
+public class JobRoleRequest {
+
+    private String roleName;
+    private String description;
+    private String sharepointUrl;
+    private String responsibilities;
+    private int numberOfOpenPositions;
+    private String location;
+    private Date closingDate;
+    private int capabilityId;
+    private int bandId;
+
+    public String getRoleName() {
+        return roleName;
+    }
+
+    public void setRoleName(final String roleName) {
+        this.roleName = roleName;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(final String description) {
+        this.description = description;
+    }
+
+    public String getSharepointUrl() {
+        return sharepointUrl;
+    }
+
+    public void setSharepointUrl(final String sharepointUrl) {
+        this.sharepointUrl = sharepointUrl;
+    }
+
+    public String getResponsibilities() {
+        return responsibilities;
+    }
+
+    public void setResponsibilities(final String responsibilities) {
+        this.responsibilities = responsibilities;
+    }
+
+    public int getNumberOfOpenPositions() {
+        return numberOfOpenPositions;
+    }
+
+    public void setNumberOfOpenPositions(final int numberOfOpenPositions) {
+        this.numberOfOpenPositions = numberOfOpenPositions;
+    }
+
+    public String getLocation() {
+        return location;
+    }
+
+    public void setLocation(final String location) {
+        this.location = location;
+    }
+
+    public Date getClosingDate() {
+        return closingDate;
+    }
+
+    public void setClosingDate(final Date closingDate) {
+        this.closingDate = closingDate;
+    }
+
+    public int getCapabilityId() {
+        return capabilityId;
+    }
+
+    public void setCapabilityId(final int capabilityId) {
+        this.capabilityId = capabilityId;
+    }
+
+    public int getBandId() {
+        return bandId;
+    }
+
+    public void setBandId(final int bandId) {
+        this.bandId = bandId;
+    }
+
+    @JsonCreator
+    public JobRoleRequest(
```

Comment:
Minor thing Niall - could you put the constructor before the getters/setters please